config-model/src/main/java/com/yahoo/vespa/model/content/cluster/ContentCluster.java

```diff
@@ -748,6 +752,31 @@ public void getConfig(BucketspacesConfig.Builder builder) {
         }
     }
 
+    @Override
+    public void getConfig(DistributionConfig.Builder builder) {
+        DistributionConfig.Cluster.Builder clusterBuilder = new DistributionConfig.Cluster.Builder();
+
+        if (redundancy() != null)
+            clusterBuilder.redundancy(redundancy().effectiveFinalRedundancy());
+
+        if (getRootGroup() != null)
+            clusterBuilder.group(getRootGroup().getGroupStructureConfig().stream()
+                                               .map(StorDistributionConfig.Group.Builder::build)
+                                               .map(config -> new DistributionConfig.Cluster.Group.Builder()
+                                                       .index(config.index())
+                                                       .name(config.name())
+                                                       .capacity(config.capacity())
+                                                       .partitions(config.partitions())
+                                                       .nodes(config.nodes().stream()
+                                                                    .map(node -> new DistributionConfig.Cluster.Group.Nodes.Builder()
+                                                                            .index(node.index())
+                                                                            .retired(node.retired()))
+                                                                    .collect(toList())))
+                                               .collect(toList()));
```

Comment:
I feel that according to our code standard you must be able to read the entire statement out loud without drawing breath or it should be split up ...

Reply:
🌬️ Barely made it! 😓 
So the idea was simple: grab a builder of the similar config, and do a deep conversion of it to this config builder type. Starting with `clusterBuilder.group(...)` you know everything inside is one group-builder, or a list of them. Streams make it clearer, I thought! I see now that the whole simple conversion thing wasn't obvious, though, since the next thing is `getRootGroup().getGroupStructureConfig()...` 😖 Fine, fine I offer you this next commit in penitence.